kitsune/users/tests/test_migration_0033.py

```diff
@@ -0,0 +1,110 @@
+
+        # Case 1: Non-migrated user with no content (should be deleted)
+        ProfileFactory(user__username="user1", is_fxa_migrated=False)
+
+        # Case 2: Non-migrated user with a question (should be kept)
+        user2 = ProfileFactory(user__username="user2", is_fxa_migrated=False).user
+        QuestionFactory(creator=user2)
+
+        # Case 3: Non-migrated user with an answer (should be kept)
+        user3 = ProfileFactory(user__username="user3", is_fxa_migrated=False).user
+        AnswerFactory(creator=user3)
+
+        # Case 4: Non-migrated user with a revision (should be kept)
+        user4 = ProfileFactory(user__username="user4", is_fxa_migrated=False).user
+        RevisionFactory(creator=user4)
+
+        # Case 5: Migrated user with no content (should be kept)
+        ProfileFactory(user__username="user5", is_fxa_migrated=True)
+
+        # Case 6: Non-migrated user with a question vote (should be kept)
+        user6 = ProfileFactory(user__username="user6", is_fxa_migrated=False).user
+        QuestionVoteFactory(creator=user6)
+
+        # Case 7: Non-migrated user with an answer vote (should be kept)
+        user7 = ProfileFactory(user__username="user7", is_fxa_migrated=False).user
+        AnswerVoteFactory(creator=user7)
+
+        # Case 8: Non-migrated user who is a sender of inbox messages (should be kept)
+        user8 = ProfileFactory(user__username="user8", is_fxa_migrated=False).user
+        InboxMessage.objects.create(to=user2, sender=user8, message="test")
+
+        # Case 9: Non-migrated user with an outbox message (should be kept)
+        user9 = ProfileFactory(user__username="user9", is_fxa_migrated=False).user
+        outbox_msg = OutboxMessage.objects.create(sender=user9, message="test")
+        outbox_msg.to.add(user2)
+
+        # Case 10: Non-migrated user who is a sender of inbox messages (should be kept)
+        user10 = ProfileFactory(user__username="user10", is_fxa_migrated=False).user
+        InboxMessage.objects.create(to=user2, sender=user10, message="test")
+
+        # Case 11: Non-migrated user with a gallery image (should be kept)
+        user11 = ProfileFactory(user__username="user11", is_fxa_migrated=False).user
+        ImageFactory(creator=user11)
+
+        # Case 12: Non-migrated user with a gallery video (should be kept)
+        user12 = ProfileFactory(user__username="user12", is_fxa_migrated=False).user
+        VideoFactory(creator=user12)
+
+        # Case 13: Non-migrated user with a forum thread (should be kept)
+        user13 = ProfileFactory(user__username="user13", is_fxa_migrated=False).user
+        ThreadFactory(creator=user13)
+
+        # Case 14: Non-migrated user with a forum post (should be kept)
+        user14 = ProfileFactory(user__username="user14", is_fxa_migrated=False).user
+        thread = ThreadFactory()
+        PostFactory(thread=thread, author=user14)
```

Comment:
Nit. You can just use `PostFactory(author=user14)` because the `PostFactory` will create the thread for you.